config/initializers/elasticsearch.rb

```diff
@@ -1,16 +1,24 @@
-yaml = YAML.load_file("#{Rails.root}/config/elasticsearch.yml").presence
+# frozen_string_literal: true
 
-Elasticsearch::Persistence.client = Elasticsearch::Client.new(log: Rails.env.development?,
-                                                              hosts: yaml['hosts'],
-                                                              user: yaml['user'],
-                                                              password: yaml['password'],
-                                                              randomize_hosts: true,
-                                                              retry_on_failure: true,
-                                                              reload_connections: true)
+module ES
+  CONFIG = YAML.load_file("#{Rails.root}/config/elasticsearch.yml").presence.freeze
+
+  def self.client
+    Elasticsearch::Client.new(log: Rails.env.development?,
+                              hosts: CONFIG['hosts'],
+                              user: CONFIG['user'],
+                              password: CONFIG['password'],
+                              randomize_hosts: true,
+                              retry_on_failure: true,
+                              reload_connections: true)
+  end
+end
 
 if Rails.env.development?
   logger = ActiveSupport::Logger.new(STDERR)
   logger.level = Logger::DEBUG
   logger.formatter = proc { |_s, _d, _p, m| "\e[2m#{m}\n\e[0m" }
-  Elasticsearch::Persistence.client.transport.logger = logger
+  ES.client.transport.logger = logger
 end
+
+Elasticsearch::Persistence.client = ES.client
```

Comment:
Why is the assignment to Elasticsearch::Persistence.client necessary? Backwards compatibility for some code that this PR doesn't change?

Reply:
Yes, that's for backwards compatibility. It sets the client for the persisted classes (`Document` and `Collection`), replicating the logic of the [existing client assignment](https://github.com/GSA/i14y/blob/master/config/initializers/elasticsearch.rb#L3), but using the new `ES.client` method. That line is going to go away in a subsequent commit.